experiments/Train.ipynb

```diff
@@ -159,380 +150,840 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 37.5 s, sys: 16.1 s, total: 53.6 s\n",
+      "Wall time: 8.39 s\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "4222"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "\n",
+    "feature_name = 'ingredients_tags'\n",
+    "\n",
+    "ingredients_input = tf.keras.Input(shape=(None,), dtype=tf.string, name=feature_name)\n",
     "\n",
-    "config=dacite.from_dict(Config, json_config)\n",
-    "model_config=config.model_config\n",
+    "ingredients_vocab = get_vocabulary(\n",
+    "    flat_batch(select_feature(ds, feature_name), batch_size=PREPROC_BATCH_SIZE),\n",
+    "    min_freq = model_config.ingredient_min_count)\n",
     "\n",
-    "output_dir:pathlib.Path = pathlib.Path(\"../models\")\n",
-    "replicates = 1\n",
+    "ingredients_graph = layers.StringLookup(\n",
+    "    vocabulary = ingredients_vocab,\n",
+    "    output_mode = 'multi_hot')(ingredients_input)\n",
     "\n",
-    "output_dir.mkdir(parents=True, exist_ok=True)"
+    "inputs[feature_name] = ingredients_input\n",
+    "input_graphs[feature_name] = ingredients_graph\n",
+    "\n",
+    "len(ingredients_vocab)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "eb92b58e",
+   "id": "db8a30f1",
    "metadata": {},
    "source": [
-    "# Preprocess training data"
+    "## Model output"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 7,
-   "id": "8c68cf1a",
-   "metadata": {
-    "scrolled": false
-   },
+   "execution_count": 10,
+   "id": "1d3426e3",
+   "metadata": {},
    "outputs": [
     {
-     "name": "stderr",
+     "name": "stdout",
      "output_type": "stream",
      "text": [
-      "2022-04-21 10:33:28.549485: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA\n",
-      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
-      "2022-04-21 10:33:28.794152: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)\n"
+      "CPU times: user 35.2 s, sys: 15.9 s, total: 51.1 s\n",
+      "Wall time: 7.95 s\n"
      ]
     },
+    {
+     "data": {
+      "text/plain": [
+       "3969"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "\n",
+    "labels = 'categories_tags'\n",
+    "\n",
+    "categories_vocab = get_vocabulary(\n",
+    "    flat_batch(select_feature(ds, labels), batch_size=PREPROC_BATCH_SIZE),\n",
+    "    min_freq = model_config.category_min_count)\n",
```

Comment:
Hum, this parameter should have a comment. I think it can really change a lot performances.
If I wanted to cheat, I would make this parameter up ;-)